int32 tracking_time
---
float range
---
bool end